src/js/nightly.js

```diff
@@ -64,89 +63,99 @@ function populateNightly() {
+
+            // update the HTML container element with this new, blank, template row (hidden at this stage)
+            nightlyList.innerHTML = newNightlyContent;
+
+            // set variables for HTML elements.
+            var dlButton = document.getElementById("nightly-dl"+tableRowCounter);
+            //var dlContent = document.getElementById("nightly-dl-content"+tableRowCounter);
+
+            // populate the new entry with that release's information
+            var publishedAt = (releasesJson[nightlyReleaseCounter].published_at);
+            document.getElementById("nightly-release"+tableRowCounter).innerHTML = (releasesJson[nightlyReleaseCounter].name).slice(0, 12); // the release name, minus the timestamp
+            document.getElementById("nightly-release"+tableRowCounter).href = ("https://github.com/AdoptOpenJDK/openjdk-nightly/releases/tag/" + releasesJson[nightlyReleaseCounter].name) // the link to that release on GitHub
+            document.getElementById("nightly-date"+tableRowCounter).innerHTML = moment(publishedAt).format('Do MMMM YYYY'); // the timestamp converted into a readable date
+            //document.getElementById("nightly-changelog"+tableRowCounter).href = releasesJson[nightlyReleaseCounter].name; // TODO: WAITING FOR THE LINKS TO BE AVAILABLE. the link to the release changelog
+            document.getElementById("nightly-timestamp"+tableRowCounter).innerHTML = (releasesJson[nightlyReleaseCounter].name).slice(13, 25); // the timestamp section of the build name
+            //document.getElementById("nightly-buildnumber"+tableRowCounter).innerHTML = releasesJson[nightlyReleaseCounter].id; // TODO: currently this is the release ID
+            //document.getElementById("nightly-commitref"+tableRowCounter).innerHTML = releasesJson[nightlyReleaseCounter].name; // TODO: WAITING FOR THE INFO TO BE AVAILABLE.
+            //document.getElementById("nightly-commitref"+tableRowCounter).href = releasesJson[nightlyReleaseCounter].name; // TODO: WAITING FOR THE LINKS TO BE AVAILABLE.
+
+            // TODO: this should be abstracted - a global function should be able to search the filename for different content and return the correct platform name.
+            if(a.indexOf("S390X_LINUX") >= 0) {
+              document.getElementById("platform-block"+tableRowCounter).innerHTML = "Linux s390x";
+            } else if(a.indexOf("X64_LINUX") >= 0) {
+              document.getElementById("platform-block"+tableRowCounter).innerHTML = "Linux x86-64";
+            } else if (a.indexOf("WIN") >= 0) {
+              document.getElementById("platform-block"+tableRowCounter).innerHTML = "Windows";
+            } else if (a.indexOf("MAC") >= 0) {
+              document.getElementById("platform-block"+tableRowCounter).innerHTML = "macOS";
```

Comment:
No problem - I also have a solution in mind (that I will implement in 2 weeks' time) that works by having an array of JS objects.

Each object will represent an 'accepted platform', and will contain information about that platform, such as the 'official name' (including x86-64 etc), the associated file extension (.zip, .tar.gz, etc), and any other information we want to include.

This will ultimately be the one single place where a change is required, in order to add a new platform to the website. The rest of the JS, HTML etc. will be abstracted to deal with as many objects (platforms) as you want to put into the array.